### ISSUE TITLE
Breadcrumb context menu slow to appear when user has many local pins

### ISSUE THREAD

**Issue by lidel** [ipfs-shipyard/ipfs-webui#1618]
Something to be aware of: the more Pins user has, the slower `pin.ls` is. 
I have around 1k of pins and it takes 10 seconds to do pin check against all pinned blocks via URL executed by my browser:

```console
$ time curl -X POST 'http://127.0.0.1:5001/api/v0/pin/ls?paths=bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi&stream=true&arg=bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi'
{"Message":"path 'bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi' is not pinned","Code":0,"Type":"error"}
curl -X POST   0.00s user 0.00s system 0% cpu 10.774 total
```

This is result of inefficiencies in go-ipfs 0.6 (pinstore needs revamp, I use flatfs on slow spinning disk) but illustrates that if we block UI on pin status check, it may take 10 seconds or more. 

I suggest we are cognizant of this possibility and audit our UI, decouppling "realtime" elements  (breadcrumbs context menu – #1599) from `pin ls` for snappier interface. 

For example, when pinning service integration lands (https://github.com/ipfs/ipfs-gui/issues/91), we could have static action in context menus to open "pin status modal" and fetch pin status (local+remote) only there.

**Comment by lidel** [comment]
I believe its slow because you do `pin ls` against all pinned blocks (including indirect), and from the perspective of GUI we only care about the root CID which was pinned recursively. 


If you add `type: recursive` filter to `pin ls`, the check is instant:

```console
$ time curl -X POST 'http://127.0.0.1:5001/api/v0/pin/ls?paths=bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi&stream=true&arg=bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi&type=recursive'
{"Message":"path 'bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi' is not pinned","Code":0,"Type":"error"}
curl -X POST   0.00s user 0.00s system 90% cpu 0.007 total
```

@rafaelramalho19 Mind opening a PR with a fix for context menus (both breadcrumbs and ones in file list), so we can include it in v2.11?

**Comment by lidel** [comment]
Fixed by #1619
